### PR TITLE
Refactor BurstDelay to BurstDelays, supporting per-burst delays

### LINKED ISSUE #13683: Support per-burst BurstDelays
Original TS engine 2.x (2.x patch, Firestorm) added support for BurstDelay0-4, allowing to define individual delays for bursts 2-6.

In our case, I'd turn `int BurstDelay` into `int[] BurstDelays`. For individual delays modders will have to define the whole list of delays (i.e. if `Burst: 6` where delay between 2nd and 3rd and 4th & 5th should be 10 and 5 between all other bursts, it would have to be `BurstDelays: 5, 10, 5, 10, 5`).
For convenience, if only a single value is defined it will be used for all bursts.

I have a branch for this, but it's based and therefore depends on #13597.

Needed to bring Juggernaut even closer to original (first 2 bursts almost simultaneously, 3rd with some more delay).

### PR DESCRIPTION
**Pull request by reaperrr**
Closes #13683.

~Depends on #13597.~

### CODE REVIEW

**Comment by reaperrr** [comment]
Rebased.

**Comment by reaperrr** [comment]
Rebased.

**Review by pchote — APPROVED** [review]

File: OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs [hunk 1]
```diff
@@ -1092,6 +1092,15 @@ internal static void UpgradeWeaponRules(ModData modData, int engineVersion, ref
 					}
 				}
 
+				// Rename BurstDelay to BurstDelays
+				if (engineVersion < 20170716)
+				{
+					if (node.Key == "BurstDelay")
+						node.Key = "BurstDelays";
+					if (node.Key == "-BurstDelay")
```
Comment (rob-v) [inline OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs]: It looks unnecessary/wrong to me. Isn't `-Name`... applied only for Traits and not properties?

**Review by rob-v — COMMENTED** [review]
One issue? + Missing rename: https://github.com/OpenRA/OpenRA/blob/bleed/mods/ra/weapons/missiles.yaml#L53

File: OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs (hunk 1 above)
Comment (reaperrr) [inline OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs]: We do have a few cases in our official mods where inherited properties are cleared/reset to default this way, but I guess you're right that it's unnecessary here, for BurstDelay specifically it's extremely unlikely.

**Comment by reaperrr** [comment]
Updated.

File: OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs [hunk 2]
```diff
@@ -1092,6 +1092,13 @@ internal static void UpgradeWeaponRules(ModData modData, int engineVersion, ref
 					}
 				}
 
+				// Rename BurstDelay to BurstDelays
+				if (engineVersion < 20170818)
+				{
+					if (node.Key == "BurstDelay")
+						node.Key = "BurstDelays";
+				}
```
Comment (rob-v) [inline OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs]: style nit: Usually without {} in this case.

**Review by rob-v — APPROVED** [review]
👍

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/92f2f9543dee21291f40e5dcc19d85061f17a73c...2fa9d836d863348a7c58e6715b5f4d87035f9edf

**Pull request merged**